### PR TITLE
Add NVDefineSpaceNew function.

### PR DESCRIPTION
**Pull request by ChriMarMe**
Supports NVPublic structure as argument.

Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: tpm2/tpm2.go [hunk 1]
```diff
@@ -1211,6 +1211,32 @@ func NVDefineSpace(rw io.ReadWriter, owner, handle tpmutil.Handle, ownerAuth, au
 	return err
 }
```
Comment (chrisfenner) [inline tpm2/tpm2.go:1212]: Do you think you could eliminate encodeDefineSpace by implementing NVDefineSpace on top of NVDefineSpaceNew?

File: tpm2/tpm2.go [hunk 2]
```diff
@@ -1211,6 +1211,32 @@ func NVDefineSpace(rw io.ReadWriter, owner, handle tpmutil.Handle, ownerAuth, au
...
 
+// NVDefineSpaceNew accepts NVPublic structure and is more flexible.
+func NVDefineSpaceNew(rw io.ReadWriter, owner tpmutil.Handle, ownerAuth, authVal string, pubInfo NVPublic) error {
```
Comment (chrisfenner) [inline tpm2/tpm2.go:1215]: Nit: Would NVDefineSpaceEx be a better name?

File: tpm2/tpm2.go [hunk 3]
```diff
@@ -1211,6 +1211,32 @@ func NVDefineSpace(rw io.ReadWriter, owner, handle tpmutil.Handle, ownerAuth, au
 	return err
 }
 
+// NVDefineSpaceNew accepts NVPublic structure and is more flexible.
```
Comment (chrisfenner) [inline tpm2/tpm2.go:1214]: Should there be a test added that exercises this new function?

Comment (ChriMarMe) [reply tpm2/tpm2.go]: Old test covers this.

**Pull request merged**